src/helpers/song_selection.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Returns a list of songs from the data store, narrowed down by the specified game options
+ * @param guildPreference - The GuildPreference
+ * @param ignoredSongs - List of YouTube video IDs of songs to ignore
+ * @returns a list of songs, as well as the number of songs before the filter option was applied
+ */
+
+import dbContext from "../database_context";
+import { Gender } from "../commands/game_options/gender";
+import GuildPreference from "../structures/guild_preference";
+import { QueriedSong } from "../types";
+import { SubunitsPreference } from "../commands/game_options/subunits";
+import { ArtistType } from "../commands/game_options/artisttype";
+import { LanguageType } from "../commands/game_options/language";
+
+// eslint-disable-next-line import/prefer-default-export
+export async function getFilteredSongList(guildPreference: GuildPreference, ignoredSongs?: Set<string>, alternatingGender?: Gender): Promise<{ songs: QueriedSong[], countBeforeLimit: number }> {
```

Comment:
I don't think this file is used anywhere/what is the purpose of having this function with a few changes from the one in game_utils?